src/GraphQL.Client.Serializer.SystemTextJson/PolymorphicTypenameConverter.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace GraphQL.Client.Serializer.SystemTextJson
+{
+    public abstract class PolymorphicTypenameConverter<T> : JsonConverter<T>

```

Comment:
Perhaps we can find a class name which makes the use case for this converter a little bit more obvious (in that it's tied to the builtin GraphQL `__typename` field).

Reply:
`FromTypenameConverter` ?

Reply:
`FromGraphQLTypenameConverter`? 😉